src/bp/core/services/stats/index.ts

```diff
@@ -0,0 +1,125 @@
+import axios from 'axios'
+import LicensingService from 'common/licensing-service'
+import { machineUUID } from 'common/stats'
+import { TYPES } from 'core/types'
+import { inject, injectable } from 'inversify'
+import ms from 'ms'
+import path from 'path'
+
+import { GhostService } from '..'
+import { BotService } from '../bot-service'
+import { JobService } from '../job-service'
+
+const LOCK_RESOURCE = 'botpress:statsService'
+@injectable()
+export class StatsService {
+  constructor(
+    @inject(TYPES.JobService) private jobService: JobService,
+    @inject(TYPES.BotService) private botService: BotService,
+    @inject(TYPES.GhostService) private ghostService: GhostService,
+    @inject(TYPES.LicensingService) private licenseService: LicensingService
+  ) {}
+
+  public start() {
+    // tslint:disable-next-line: no-floating-promises
+    this.run()
+    setInterval(this.run.bind(this), ms('6 hours'))
+  }
+
+  private async run() {
+    const lock = await this.jobService.acquireLock(LOCK_RESOURCE, ms('1 minute'))
+    if (lock) {
+      await this.sendStats()
+    }
+  }
+
+  private async sendStats() {
+    const stats = await this.getStats()
+    try {
+      await axios.post('https://telemetry.botpress.io/ingest', stats)
+    } catch (err) {
+      // silently fail
+    }
+  }
+
+  private async getStats() {
+    return {
+      schema: '1.0.0',
+      timestamp: new Date().toISOString(),
+      bots: {
+        count: await this.getBotsCount()
+      },
+      flows: {
+        count: await this.getFlowCount()
+      },
+      intents: {
+        count: await this.getIntentsCount()
+      },
+      contentElements: {
+        count: await this.getContentElementsCount()
+      },
+      server: {
+        externalUrl: process.EXTERNAL_URL || `http://${process.HOST}:${process.PORT}`,
```

Comment:
EXTERNAL_URL will always be set no matter what, so || not necessary. What about LOCAL_URL ? Could be nice to also add IS_PRO_AVAILABLE for the type of build